hosted-cluster-config-operator/controllers/resources/resources.go

```diff
@@ -73,14 +92,47 @@ func (r *reconciler) reconcile(ctx context.Context) error {
 	log := ctrl.LoggerFrom(ctx)
 	log.Info("Reconciling")
 
+	hcp := manifests.HostedControlPlane(r.hcpNamespace, r.hcpName)
+	if err := r.cpClient.Get(ctx, crclient.ObjectKeyFromObject(hcp), hcp); err != nil {
+		return fmt.Errorf("failed to get hosted control plane %s/%s: %w", r.hcpNamespace, r.hcpName, err)
+	}
+
+	globalConfig, err := globalconfig.ParseGlobalConfig(ctx, hcp.Spec.Configuration)
+	if err != nil {
+		return fmt.Errorf("failed to parse global config for control plane %s/%s: %w", r.hcpNamespace, r.hcpName, err)
+	}
+
+	var errs []error
+	log.Info("reconciling guest cluster crds")
+	if err := r.reconcileCRDs(ctx); err != nil {
```

Comment:
I like the idea with collecting the errors and try to get as much done as possible before returning them. However, this is one of the things that is almost guaranteed to accidentally get broken in the future if we don't test that behavior.

How about a unittest that:
1. Runs the `Reconcile` and counts the number of resources created across all types
2. Changes the client to use an empty one that randomly injects errors when `Geting` resources and counts how often that happened
3. Run `Reconcile` again and assert that the number of resources created is the number from step 1 minues the number of errors injected
4. Repeated 2+3 a 100 times to be relatively sure we catch any errors